src/responder/common/cache_req/cache_req_search.c

```diff
@@ -325,9 +361,16 @@ cache_req_search_send(TALLOC_CTX *mem_ctx,
          * we can return it without searching the rest of the domains.
          */
         if (status != CACHE_OBJECT_MISSING) {
-            CACHE_REQ_DEBUG(SSSDBG_TRACE_FUNC, cr,
-                            "Object found, but needs to be refreshed.\n");
-            bypass_dp = false;
+            if (cache_req_should_be_in_cache (cr, state->result)) {
```

Comment:
I would personally suggest to add this check to `cache_req_search_cache` directly and if the check "matches" (so, if the entry is supposed to be filtrered out), I would just pass `ERR_ID_OUTSIDE_RANGE` to the big switch in `cache_req_search_cache`.

And another potential idea: some cache_req searches might return more than one entry. The extreme case is enumeration which returns all. Should we in that case filter out only a subset those entries that match the ID constrains? If yes, then maybe it would be even better to have two or three common "checks" and assign them to each cache_req plugin as a new cache_req plugin property.

Reply:
> I would personally suggest to add this check to `cache_req_search_cache` directly and if the check "matches" (so, if the entry is supposed to be filtrered out), I would just pass `ERR_ID_OUTSIDE_RANGE` to the big switch in `cache_req_search_cache`.
> 

If we move this check into  `cache_req_search_cache` it will be performed on every request. 
This way it is performed only on expired items => less often. Returning `ERR_ID_OUTSIDE_RANGE` is good idea, PR updated.

> And another potential idea: some cache_req searches might return more than one entry. The extreme case is enumeration which returns all. Should we in that case filter out only a subset those entries that match the ID constrains? If yes, then maybe it would be even better to have two or three common "checks" and assign them to each cache_req plugin as a new cache_req plugin property.

Few lines above we call `cache_req_expiration_status` which should add expiration timestamp. This function also assumes that there is only one (or none) message in the response. I guess it is OK and it is handled somewhere else. But I'm not sure about this.

@pbrezina what do you think?

Reply:
> Shouldn't we return ERR_ID_OUTSIDE_RANGE for both expired and _not_ expired objects?

Yes, we can. 

- + we will react on config change immediately.
- - we will be checking always all requested objects

Just what is more important. I thing that looking inside every object is not so expensive but have no experience with sssd in large scale deployment. 
@pbrezina, @jhrozek  - would it be fine to check all the time? In the bug customer expects that object will expire after its TTL so it will be fine for him like this.